### ISSUE TITLE
Multiple "Any tag is" (negated or not) rule triggers don't all apply in strict mode

### ISSUE THREAD

**Issue by CtrlZvi** [firefly-iii/firefly-iii#8291]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Creating a rule that has two (or more) "Any tag is .." triggers (whether or not they are negated) does not work as expected in strict mode.

### Debug information

Debug information generated at 2023-12-24 09:35:01 for Firefly III version **v6.1.0**.

<table><thead><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Firefly III</td><td>v6.1.0 / v2.0.12 / 22 (exp. 22)</td></tr><tr><td>PHP version</td><td>8.3.0 / apache2handler / Linux</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Error reporting</td><td>Display: Off, reporting: ALL errors</td></tr><tr><td>Max upload</td><td>67108864 (64 MB)</td></tr><tr><td>Database drivers</td><td><strong>*mysql*</strong>, pgsql, sqlite, </td></tr><tr><td>Docker build</td><td><span>#</span>852, base <span>#</span>67</td></tr></tbody></table><table><thead><tr><th colspan="2">Firefly III information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Timezone</td><td>Europe/Amsterdam + America/Los_Angeles</td></tr><tr><td>App environment</td><td>local, debug: false</td></tr><tr><td>Layout</td><td>v1</td></tr><tr><td>Logging</td><td>notice, stack / (empty)</td></tr><tr><td>Cache driver</td><td>file</td></tr><tr><td>Default language and locale</td><td>en_US + equal</td></tr><tr><td>Trusted proxies</td><td></td></tr><tr><td>Login provider & user guard</td><td>eloquent / web</td></tr><tr><td>Login headers</td><td>N/A + N/A</td></tr><tr><td>Stateful domains</td><td></td></tr><tr><td>Last cron job</td><td>never (never)</td></tr><tr><td>Mailer</td><td>log</td></tr></tbody></table><table><thead><tr><th colspan="2">User-specific information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>User</td><td><span>#</span>1 of 1</td></tr><tr><td>User flags</td><td><span title="Has liabilities">:credit_card:</span> <span title="Has reconciled">:ledger:</span> <span title="Has rules">:wrench:</span> <span title="Has object groups">:bookmark_tabs:</span> <span title="Has subscriptions">:email:</span></td></tr><tr><td>Session start</td><td>2023-11-01 00:00:00</td></tr><tr><td>Session end</td><td>2023-11-30 09:21:17</td></tr><tr><td>View range</td><td>1M</td></tr><tr><td>User language</td><td>en_US</td></tr><tr><td>User locale</td><td>en_US</td></tr><tr><td>Locale(s) supported</td><td>en_US.utf8: :white_check_mark:<br>en_US.UTF-8: :white_check_mark:<br></td></tr><tr><td>User agent</td><td>Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:121.0) Gecko/20100101 Firefox/121.0</td></tr></tbody></table>

### Expected behaviour

Each of the "Any tag is .." triggers must be correctly matched for the transaction to have the rules applied. Instead, it seems that only the first non-negated trigger applies, and any after are effectively ignored.

### Steps to reproduce

1. Create a transaction with only the "Test Tag 1" tag
2. Create a transaction with only the "Test Tag 2" tag
3. Create a rule with two "Any tag is .." triggers
    1. The first trigger should be for "Test Tag 1"
    2. The second trigger should be for "Test Tag 2"
4. Have the rule action set the category to "Test Category"
4. Run the rule and see that both transactions are in "Test Category" instead of only the second transaction

### Additional info

The bug seems to also interact weirdly with negated versions as well. My tests with exactly one negated "Any tag is.." rule before the first un-negated one respected the negated trigger, but if it was after an un-negated trigger, it was ignored.

**Comment by JC5** [comment]
Thanks, I'll check it out!

**Comment by JC5** [comment]
Alright check out the `:develop` tag when you're able, I think I fixed this. The new image should be ready in 30m.

**Comment by CtrlZvi** [comment]
Not sure if I tested correctly (I just updated my docker-compose.yml to use the `:develop` tag instead of latest and then destroyed the containers and reran), but now any searches that involve tags seem to return no results (whether part of a rule or not).

**Comment by CtrlZvi** [comment]
From the logs, it seems like there's some lowercasing going on that's causing match failures:
```
firefly_iii_core | 192.168.0.50 - - [24/Dec/2023:22:27:24 +0100] "GET /search?search=tag_is%3A%22fast+food%22 HTTP/1.1" 200 8030 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:121.0) Gecko/20100101 Firefox/121.0"
firefly_iii_core | 192.168.0.50 - - [24/Dec/2023:22:27:24 +0100] "GET /v1/jscript/variables?ext=.js&v=6.1.0 HTTP/1.1" 200 4031 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:121.0) Gecko/20100101 Firefox/121.0"
firefly_iii_core | [2023-12-24 22:27:24] local.DEBUG: Constructed OperatorQuerySearch  
firefly_iii_core | [2023-12-24 22:27:24] local.DEBUG: Now in parseQuery(tag_is:"fast food")  
firefly_iii_core | [2023-12-24 22:27:24] local.DEBUG: Found 1 node(s)  
firefly_iii_core | [2023-12-24 22:27:24] local.DEBUG: Now in handleSearchNode(Gdbots\QueryParser\Node\Field)  
firefly_iii_core | [2023-12-24 22:27:24] local.DEBUG: Now handle Node class Gdbots\QueryParser\Node\Field  
firefly_iii_core | [2023-12-24 22:27:24] local.DEBUG: Now in updateCollector("tag_is", "fast food")  
firefly_iii_core | [2023-12-24 22:27:24] local.DEBUG: "tag_is" is not an alias.  
firefly_iii_core | [2023-12-24 22:27:24] local.DEBUG: "tag_is" is not an alias.  
firefly_iii_core | [2023-12-24 22:27:24] local.DEBUG: Added operator type "tag_is"  
firefly_iii_core | [2023-12-24 22:27:24] local.DEBUG: Now in parseTagInstructions()  
firefly_iii_core | [2023-12-24 22:27:24] local.DEBUG: 1 include tag(s)  
firefly_iii_core | [2023-12-24 22:27:24] local.DEBUG: Include tag "Fast Food"  
firefly_iii_core | [2023-12-24 22:27:24] local.DEBUG: Now in setTags(1 tag(s))  
firefly_iii_core | 127.0.0.1 - - [24/Dec/2023:22:27:35 +0100] "GET /health HTTP/1.1" 200 1776 "-" "Firefly III Health Checker/2.2"
firefly_iii_core | [2023-12-24 22:27:35] local.DEBUG: GroupCollector: postFilterCollection has 1 filter(s) and 5311 transaction(s).  
firefly_iii_core | [2023-12-24 22:27:35] local.DEBUG: Applying filter...  
```
...
```
firefly_iii_core | [2023-12-24 22:27:35] local.DEBUG: Now in setTags(Fast Food) filter  
firefly_iii_core | [2023-12-24 22:27:35] local.DEBUG: Transaction has 2 tag(s)  
firefly_iii_core | [2023-12-24 22:27:35] local.DEBUG: "data import on 2023-12-19 @ 23:18" versus ["Fast Food"] 
firefly_iii_core | [2023-12-24 22:27:35] local.DEBUG: "fast food" versus ["Fast Food"] 
firefly_iii_core | [2023-12-24 22:27:35] local.DEBUG: Found 0 tags, need at least 1. 
```

**Comment by CtrlZvi** [comment]
That seems to be [exactly it](https://github.com/firefly-iii/firefly-iii/commit/2b90c20db8f38ca6e53c7a6d07a823205441334b#diff-a9ac73c4cb8fbdb9c6888831df240dcb373d6dc07f113a846cd23ed3ef2dbf03R697).

**Comment by JC5** [comment]
Nice find, that's an easy fix. Try pulling `:develop` again.

**Comment by CtrlZvi** [comment]
Latest `:develop` seems to fix both the initial issue and the lowercasing issue. Thank you!

**Comment by JC5** [comment]
Great! I've marked it as fixed and it'll be part of 6.1.1.
